aten/src/ATen/native/mkldnn/MKLDNNConversions.cpp

```diff
@@ -198,4 +199,49 @@ Tensor mkldnn_reorder_conv3d_weight(
 
 #endif // AT_MKLDNN_ENABLED()
 
+#if AT_MKL_ENABLED()
+#include <mkl.h>
+
+Tensor reorder_linear_weight(
+    const Tensor& weight,
+    const c10::optional<int64_t> batch_size) {
+  // TODO: bf16 prepack using MKLDNN
```

Comment:
I feel it a bit confusing if we want to use a single interface to pack weight while the prepacked weight is then used by two different linear API, one for MKL and the other for MKLDNN. Would it be more clear to have separate weight packing interfaces for MKL and MKLDNN linear?